projects/portal/src/app/pages/dialogs/delegate/delegate-form-dialog/delegate-form-dialog.component.ts

```diff
@@ -56,11 +65,33 @@ export class DelegateFormDialogComponent implements OnInit {
   ngOnInit(): void {}
 
   async onSubmit($event: DelegateOnSubmitEvent) {
-    const txHash = await this.stakingAppService.createDelegate(
-      this.validator?.operator_address!,
-      $event.amount,
-      $event.minimumGasPrice,
-    );
-    this.matDialogRef.close(txHash);
+    this.validatorsList$.subscribe(async (valList) => {
+      const validatorStatus = valList?.find(
+        (val) => val.operator_address == this.validator?.operator_address,
+      )?.status;
+      if (validatorStatus != 'BOND_STATUS_BONDED') {
+        const inactiveValidatorResult = await this.dialog
+          .open(InactiveValidatorConfirmDialogComponent, {
+            data: { valAddress: this.validator?.operator_address!, isConfirmed: false },
+          })
+          .afterClosed()
+          .toPromise();
+
+        if (
+          inactiveValidatorResult === undefined ||
+          inactiveValidatorResult.isConfirmed === false
+        ) {
+          this.snackBar.open('Delegate was canceled', undefined, { duration: 6000 });
+          return;
+        }
+      }
+
+      const txHash = await this.stakingAppService.createDelegate(
+        this.validator?.operator_address!,
+        $event.amount,
+        $event.minimumGasPrice,
+      );
+      this.matDialogRef.close(txHash);
+    });
```

Comment:
@Senna46 
Sorry, I missed that.

For validatorsList$, if new data flows in, e.g. a different node is selected, is there a possibility that a transaction that has already been executed will be executed again?

I think this can be resolved by separating create this.validatorsList$ process to parent component.